server/app/controllers/lists.js

```diff
@@ -0,0 +1,134 @@
+const ErrorResponse = require('../utils/errorResponse');
+const asyncHandler = require('../middleware/async');
+const List = require('../models/List');
+
+/**
+ * @api {get} api/v1/users/:userId/lists
+ * @apiName getLists
+ * @apiGroup lists
+ *
+ * @apiDescription Get all of the lists for a user
+ */
+// eslint-disable-next-line no-unused-vars
+const getLists = asyncHandler(async (req, res, next) => {
+  const { userId } = req.params;
+  const lists = await List.find({ user: userId });
+  if (!lists) {
+    return next(
+      new ErrorResponse(
+        `No lists with the user id of ${req.params.userId}`,
+        404,
+      ),
+    );
+  }
+  return res.status(200).json({ success: true, lists });
+});
+
+/**
+ * @api {get} api/v1/users/:userId/lists/:listId
+ * @apiName getList
+ * @apiGroup lists
+ *
+ * @apiDescription Get one list for the user
+ */
+// eslint-disable-next-line no-unused-vars
+const getList = asyncHandler(async (req, res, next) => {
+  const { listId } = req.params;
+  const lists = await List.findById(listId);
+  if (!lists) {
+    return next(
+      new ErrorResponse(`No lists with the id of ${req.params.listId}`, 404),
+    );
+  }
+  return res.status(200).json({ success: true, lists });
+});
+
+/**
+ * @api {post} /api/v1/lists
+ * @apiName postList
+ * @apiGroup lists
+ * @apiPermission protected
+ *
+ * @apiDescription Creates user in database
+ */
+// eslint-disable-next-line no-unused-vars
+const postList = asyncHandler(async (req, res, next) => {
+  const { name, coverUrl, user, products } = req.body;
```

Comment:
I think here and below, you would take an actual image file instead of a url, which you would then use the s3 service (I think Simon is working on it) to upload it and get that url